File: 2017/15/15.py
#!/usr/bin/env python3


def run(val_a, val_b):
    matches = 0
    for _ in range(40_000_000):  # Way too slow
        val_a = 16807*val_a % 0x7FFF_FFFF
        val_b = 48271*val_b % 0x7FFF_FFFF
        matches += val_a&0xFFFF == val_b&0xFFFF
        continue
    return matches

assert(run(65, 8921) == 588)
print('Part 1:', run(618, 814))  # 577

File: 2017/15/15.c
#include <assert.h>
#include <stdint.h>
#include <stdio.h>

int p1(uint32_t a, uint32_t b) {
    int matches = 0;
    for (int i = 0; i < 40000000; i++) {
        a = 16807ul*a % 0x7FFFFFFF;
        b = 48271ul*b % 0x7FFFFFFF;
        matches += (a&0xFFFF) == (b&0xFFFF);
    }
    return matches;
}

int main() {
    assert(p1(65, 8921) == 588);
    printf("Part 1: %d\n", p1(618, 814));
    return 0;
}
